### ISSUE TITLE
Conv2D returns empty tensor of type float32 in case of bfloat16 input.

### ISSUE THREAD

**Issue by Poiu19** [tensorflow/tensorflow#56680]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9.1

### Custom Code

No

### OS Platform and Distribution

Linux

### Mobile device

Linux Ubuntu 20.04

### Python version

3.8

### Bazel version

_No response_

### GCC/Compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current Behaviour?

```shell
Conv2D returns empty tensor of type float32 in case of bfloat16 input.
```


### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np

for dtype in [tf.float32, tf.bfloat16]:

    strides = tf.constant([1, 1, 1, 1], dtype=tf.int32)

    input =   tf.constant(np.full(fill_value=1., shape=[1, 56, 56, 1]), dtype=dtype)
    filters = tf.constant(np.full(fill_value=1., shape=[1, 56, 1, 1]), dtype=dtype)

    padding = "SAME"

    output = tf.nn.conv2d(input, filters, strides, padding)

    print("")
    print("dtype:        ", dtype)
    print("Input size:   ", input.shape.num_elements())
    print("Output size:  ", output.shape.num_elements())
    print("Output dtype: ", output.dtype)
    assert output.shape.num_elements() > 0
```


### Relevant log output

```shell
dtype:         <dtype: 'float32'>
Input size:    3136
Output size:   3136
Output dtype:  <dtype: 'float32'>

dtype:         <dtype: 'bfloat16'>
Input size:    3136
Output size:   0
Output dtype:  <dtype: 'float32'>
collected 0 items / 1 errors                                                                                                                                                                                                          

=============================================================================================================== ERRORS ================================================================================================================
______________________________________________________________________________________________________ ERROR collecting repro.py ______________________________________________________________________________________________________
repro.py:20: in <module>
    assert output.shape.num_elements() > 0
E   assert 0 > 0
E    +  where 0 = <bound method TensorShape.num_elements of TensorShape([0])>()
E    +    where <bound method TensorShape.num_elements of TensorShape([0])> = TensorShape([0]).num_elements
E    +      where TensorShape([0]) = <tf.Tensor: shape=(0,), dtype=float32, numpy=array([], dtype=float32)>.shape
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================================================= 1 error in 2.33 seconds =======================================================================================================
```
</details>

**Comment by tilakrayal** [comment]
@Poiu19,
This issue was resolved in the latest tf-nightly version. Could you please find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/4d3d1b8c9bbb8fc23e74b7dd4f4079f6/56680.ipynb) and confirm the same. Thank you!

**Comment by Poiu19** [comment]
@tilakrayal 
The output type is now correct, but returned tensor is still empty. In the documentation I see that such a case (bfloat16 input) is supported so the output shouldn't be empty.

**Comment by tilakrayal** [comment]
@Poiu19,
The input shape is `shape=(0,)` which is not able to fetch the data. Could you please find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/9fa63ac931b6a3f76652dc2584cef9c5/untitled420.ipynb) and provide the input shape.  Please find the reference [gist](https://colab.research.google.com/gist/tilakrayal/db61f5d4b9cf2bcfe718bb3971f5aafd/untitled422.ipynb) of tf.float32 on tensoflow v2.8.
```
2.10.0-dev20220708
type:         tf.Tensor([], shape=(0,), dtype=bfloat16)
```
Thank you!

**Comment by Poiu19** [comment]
> @Poiu19, The input shape is `shape=(0,)` which is not able to fetch the data. Could you please find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/9fa63ac931b6a3f76652dc2584cef9c5/untitled420.ipynb) and provide the input shape. Please find the reference [gist](https://colab.research.google.com/gist/tilakrayal/db61f5d4b9cf2bcfe718bb3971f5aafd/untitled422.ipynb) of tf.float32 on tensoflow v2.8.
> 
> ```
> 2.10.0-dev20220708
> type:         tf.Tensor([], shape=(0,), dtype=bfloat16)
> ```
> 
> Thank you!

@tilakrayal 

The input shape is not `shape=(0,)`. What you have indicated is the output shape, take a look at what is being printed:
```
    print("type:        ", output)
```
In fact, the input shape is `shape=(1, 56, 56, 1)` and in the test with `dtype=float32`, the output has the same shape as the input, while in the test with `dtype=bfloat16` the output is `shape=(0,)`

**Comment by tilakrayal** [comment]
@sachinprasadhs,
I was able to get the `dtype:bfloat16` on tf-nightly. But I was not able to fetch the output data. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/4d3d1b8c9bbb8fc23e74b7dd4f4079f6/56680.ipynb).

**Comment by Poiu19** [comment]
bump

**Comment by cantonios** [comment]
Ugh, this is the same as #55738.  It should now throw an error with the latest nightly.

@penpornk what's the tag or contact for intel MKL?  This is because of the no-op registrations.  Intel needs to add proper kernels if they're going to be registering ops.

**Comment by cantonios** [comment]
Should be fixed with [dc3a847](https://github.com/tensorflow/tensorflow/commit/dc3a847a17078d86dcc2c13ada1cc9b9219fd662).
